asv_bench/benchmarks/benchmarks.py

```diff
@@ -11,159 +11,174 @@
+    def time_groupby_count(self, data_size):
+        self.df.groupby(by=self.groupby_column).count()
 
-    def time_groupby_mean(self, impl, data_type, data_size):
-        self.df.groupby(by=self.df.columns[0]).mean()
+    def time_groupby_size(self, data_size):
+        self.df.groupby(by=self.groupby_column).size()
 
-    def time_groupby_count(self, impl, data_type, data_size):
-        self.df.groupby(by=self.df.columns[0]).count()
+    def time_groupby_sum(self, data_size):
+        self.df.groupby(by=self.groupby_column).sum()
+
+    def time_groupby_mean(self, data_size):
+        self.df.groupby(by=self.groupby_column).mean()
 
 
 class TimeJoin:
-    param_names = ["impl", "data_type", "data_size", "how", "sort"]
+    param_names = ["data_size", "how", "sort"]
     params = [
-        ["modin", "pandas"],
-        ["int"],
-        JOIN_DATA_SIZE,
-        ["left", "right", "outer", "inner"],
-        [False, True],
+        BINARY_OP_DATA_SIZE,
+        ["left", "inner"],
+        [False],
     ]
 
-    def setup(self, impl, data_type, data_size, how, sort):
+    def setup(self, data_size, how, sort):
         self.df1 = generate_dataframe(
-            impl, data_type, data_size[0], data_size[1], RAND_LOW, RAND_HIGH
+            ASV_USE_IMPL, "int", data_size[1], data_size[0], RAND_LOW, RAND_HIGH
         )
         self.df2 = generate_dataframe(
-            impl, data_type, data_size[2], data_size[3], RAND_LOW, RAND_HIGH
+            ASV_USE_IMPL, "int", data_size[3], data_size[2], RAND_LOW, RAND_HIGH
         )
 
-    def time_join(self, impl, data_type, data_size, how, sort):
+    def time_join(self, data_size, how, sort):
         self.df1.join(
             self.df2, on=self.df1.columns[0], how=how, lsuffix="left_", sort=sort
-        )
+        ).shape
 
 
 class TimeMerge:
-    param_names = ["impl", "data_type", "data_size", "how", "sort"]
+    param_names = ["data_size", "how", "sort"]
     params = [
-        ["modin", "pandas"],
-        ["int"],
-        MERGE_DATA_SIZE,
-        ["left", "right", "outer", "inner"],
-        [False, True],
+        BINARY_OP_DATA_SIZE,
+        ["left", "inner"],
+        [False],
     ]
 
-    def setup(self, impl, data_type, data_size, how, sort):
+    def setup(self, data_size, how, sort):
         self.df1 = generate_dataframe(
-            impl, data_type, data_size[0], data_size[1], RAND_LOW, RAND_HIGH
+            ASV_USE_IMPL, "int", data_size[1], data_size[0], RAND_LOW, RAND_HIGH
         )
         self.df2 = generate_dataframe(
-            impl, data_type, data_size[2], data_size[3], RAND_LOW, RAND_HIGH
+            ASV_USE_IMPL, "int", data_size[3], data_size[2], RAND_LOW, RAND_HIGH
         )
 
-    def time_merge(self, impl, data_type, data_size, how, sort):
-        self.df1.merge(self.df2, on=self.df1.columns[0], how=how, sort=sort)
+    def time_merge(self, data_size, how, sort):
+        self.df1.merge(self.df2, on=self.df1.columns[0], how=how, sort=sort).shape
```

Comment:
I think we should make result materialization in some more obvious way, for example by adding decorator to every `time_*` method that will access `.shape` on the return value of `time_*`